pkg/testing/tools/estools/elasticsearch.go

```diff
@@ -488,7 +488,7 @@ func GetLogsForDatasetWithContext(ctx context.Context, client elastictransport.I
 		},
 	}
 
-	return performQueryForRawQuery(ctx, indexQuery, "*ds-logs*", client)
```

Comment:
This must have been querying the underlying data stream backing indexes directly then? That wasn't correct if so.

Everything follows the data stream naming scheme so `log*` is correct if you are querying for all logs data streams.

In our own tests for elastic agent logs you should be able to do `logs-elastic_agent*` but this ES instance contains such few documents I'm not sure this make much of a difference.

Reply:
We do this in other places in this file 

https://github.com/elastic/elastic-agent/blob/f7dcbd74ce0e27d3b46b478755168c66d78dc111/pkg/testing/tools/estools/elasticsearch.go#L437

https://github.com/elastic/elastic-agent/blob/f7dcbd74ce0e27d3b46b478755168c66d78dc111/pkg/testing/tools/estools/elasticsearch.go#L365

Reply:
I also see this filters by dataset already which should catch bugs where the data exists but in the wrong index, but limiting the query only to the index things are supposed to be in would be a stricter test.

Reply:
Thanks Craig! I'll update it to be more strict then.